### PR TITLE
Merge main to v3

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Merges `main` to `v3`

### Drive-by changes

- Disables sealevel in testnet4
- Disables fast token router tests
- 

### Backward compatibility

- No, testnet3 removed

### Testing

CI

### CODE REVIEW

File: typescript/infra/config/aggregationIsm.ts [hunk 1]
```diff
@@ -1,103 +1,28 @@
+// Aggregation (1/2)
```
Comment (tkporter) [inline typescript/infra/config/aggregationIsm.ts:1]: think this moved unintentionally

and actually, I removed this file in favor of the new `routingIsm`

File: typescript/infra/config/environments/mainnet2/agent.ts [hunk 2]
```diff
@@ -48,33 +50,75 @@ const contextBase = {
...
   {
     type: GasPaymentEnforcementPolicyType.None,
     // To continue relaying interchain query callbacks, we whitelist
     // all messages between interchain query routers.
     // This whitelist will become more strict with
     // https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1605
-    matchingList: interchainQueriesMatchingList,
+    matchingList: [
+      ...interchainQueriesMatchingList,
+      {
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet2/agent.ts:89]: we can remove all these warp routes & matching list changes if we want - or we can keep them in, and just remove when we deploy mainnet3, either way

File: typescript/infra/config/environments/testnet4/agent.ts [hunk 3]
```diff
@@ -57,7 +57,28 @@ const gasPaymentEnforcement: GasPaymentEnforcementConfig[] = [
     // all messages between interchain query routers.
     // This whitelist will become more strict with
     // https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1605
-    matchingList: interchainQueriesMatchingList,
+    matchingList: [
```
Comment (tkporter) [inline typescript/infra/config/environments/testnet4/agent.ts:60]: same we can just delete these changes as we're not running these on testnet4

File: typescript/infra/helm/helloworld-kathy/templates/external-secret.yaml [hunk 4]
```diff
@@ -22,13 +22,22 @@ spec:
...
+   {{/*
+   * Always get the GCP-based key, which is used for non-EVM chains.
+   */}}
+        GCP_SECRET_OVERRIDE_{{ .Values.hyperlane.context | upper }}_{{ .Values.hyperlane.runEnv | upper }}_KEY_KATHY: {{ print "'{{ .gcp_private_key | toString }}'" }}
+   {{/*
    * For each network, create an environment variable with the RPC endpoint.
    * The templating of external-secrets will use the data section below to know how
    * to replace the correct value in the created secret.
    */}}
+<<<<<<< HEAD
```
Comment (tkporter) [inline typescript/infra/helm/helloworld-kathy/templates/external-secret.yaml:34]: !!!!!! vscode

File: typescript/infra/helm/helloworld-kathy/templates/external-secret.yaml [hunk 5]
```diff
@@ -48,8 +55,13 @@ spec:
    * For each network, load the secret in GCP secret manager with the form: environment-rpc-endpoint-network,
    * and associate it with the secret key networkname_rpc.
    */}}
+<<<<<<< HEAD
```
Comment (tkporter) [inline typescript/infra/helm/helloworld-kathy/templates/external-secret.yaml:58]: ditto

**Review by tkporter — APPROVED** [review]

File: typescript/infra/config/environments/mainnet2/agent.ts (hunk 2 above)
Comment (yorhodes) [reply typescript/infra/config/environments/mainnet2/agent.ts:89]: we havent touched mainnet2 on v3 so I will leave this

**Pull request merged**